core-bundle/src/Controller/FrontendModule/FeedReaderController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\FrontendModule;
+
+use Contao\Config;
+use Contao\CoreBundle\Exception\PageNotFoundException;
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Contao\CoreBundle\Twig\FragmentTemplate;
+use Contao\Environment;
+use Contao\ModuleModel;
+use Contao\Pagination;
+use Contao\StringUtil;
+use FeedIo\Feed;
+use FeedIo\FeedIo;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * @FrontendModule(category="miscellaneous")
+ */
+class FeedReaderController extends AbstractFrontendModuleController
+{
+    public function __construct(private FeedIo $feedIo, private LoggerInterface $logger, private CacheInterface $cache)
+    {
+    }
+
+    protected function getResponse(FragmentTemplate $template, ModuleModel $model, Request $request): Response
+    {
+        $this->initializeContaoFramework();
+
+        $feeds = [];
+
+        foreach (StringUtil::trimsplit('[\n\t ]', trim($model->feed_urls)) as $url) {
+            $feed = $this->cache->get('feed_reader_'.$model->id.'_'.md5($url), function (ItemInterface $item) use ($url, $model) {
+                $feed = new Feed();
+
+                try {
+                    $this->feedIo->read($url, $feed);
+
+                    if ($model->feed_cache > 0) {
+                        $item->expiresAfter($model->feed_cache);
+                    }
+                    $item->set($feed);
+                } catch (\Exception $exception) {
+                    $this->logger->error(sprintf('Could not read feed %s: %s', $url, $exception->getMessage()));
```

Comment:
should this somehow cancel the feed (`return`)? Looks like `$feed` might not exist?